rackcity/views/user_views.py

```diff
@@ -442,3 +447,108 @@ def usernames(request):
         {"usernames": usernames},
         status=HTTPStatus.OK,
     )
+
+
+@api_view(['POST'])
+@permission_classes([IsAdminUser])
+def user_set_groups(request):
+    """
+    Set groups for a user.
+    """
+    data = JSONParser().parse(request)
+    if 'id' not in data:
+        return JsonResponse(
+            {
+                "failure_message":
+                    Status.MODIFY_ERROR.value +
+                    GenericFailure.INTERNAL.value,
+                "errors": "Must specify user id on admin permission revoke"
+            },
+            status=HTTPStatus.BAD_REQUEST,
+        )
+    try:
+        user = User.objects.get(id=data['id'])
+    except ObjectDoesNotExist:
+        return JsonResponse(
+            {
+                "failure_message":
+                    Status.MODIFY_ERROR.value +
+                    "User" + GenericFailure.DOES_NOT_EXIST.value,
+                "errors": "No existing user with id=" + data['id']
+            },
+            status=HTTPStatus.BAD_REQUEST,
+        )
+    groups_added = []
+    groups_removed = []
+    for group in GroupName:
+        group_key = group.value
+        if group_key in data:
+            if data[group_key]:
+                added = add_user_to_group(user, group)
+                if added:
+                    groups_added.append(group_key)
+            else:
+                removed = remove_user_from_group(user, group)
+                if removed:
+                    groups_removed.append(group_key)
+    success_message = ""
+    if len(groups_added) > 0:
+        success_message += \
+            ("User added to groups: " + ", ".join(groups_added) + ". ")
```

Comment:
Do you think it'll be concerning if you add a user to Models and Assets, but they were already in Assets, so the success message only says they were added to Models?  What if the success message also said something like "User was already in groups: Assets"

Reply:
This may depend on what the workflow is like on the frontend - like if it's a multi-select you may be sending the redundant ones every time so maybe you don't want that in the success message.  Also possible you'd never have the case where you thought you added them but they were already in it

Reply:
I like that; I'll modify it

Reply:
can't believe I used a semicolon

Reply:
A period would have been fine